core/src/main/java/nu/yona/server/device/entities/DeviceAnonymized.java

```diff
@@ -37,6 +46,14 @@
 
 	private String appVersion;
 
+	@OneToMany(mappedBy = "deviceAnonymized", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
+	private Set<VpnStatusChangeEvent> vpnStatusChangeEvents;
+
+	@ManyToOne
+	@JoinColumnsOrFormulas({
+			@JoinColumnOrFormula(formula = @JoinFormula(value = "(SELECT e.id FROM vpn_status_change_events e WHERE e.device_anonymized_id = id ORDER BY e.id DESC LIMIT 1)", referencedColumnName = "id")) })
```

Comment:
Isn't it more common to order by `eventTime`?

Reply:
I've chosen for ID as always increments and is indexed already, while the resolution of the time field might be so low that duplicate values occur for events that come in quick succession.

Reply:
The resolution of the date/time might be too low to tell which one came first for two messages that came in close succession. Since the ID is auto-increment, such issues don't exist, and it's indexed already, so it's cheapest too.